recipes/nodejs/all/conanfile.py

```diff
@@ -1,48 +1,82 @@
 import os
-from conans import ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+import re
+from conans import __version__
+from conan import ConanFile
+from conan.tools.scm import Version
+from conan.tools.files import copy, get
+from conan.errors import ConanInvalidConfiguration
 
-required_conan_version = ">=1.33.0"
+from six import StringIO
+
+required_conan_version = ">=1.59.0"
 
 
 class NodejsConan(ConanFile):
     name = "nodejs"
     description = "nodejs binaries for use in recipes"
-    topics = ("conan", "node", "nodejs")
+    topics = ("node", "nodejs")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://nodejs.org"
     license = "MIT"
-    settings = "os", "arch"
+    settings = "os", "arch", "compiler"
     no_copy_source = True
+    short_paths = True
 
     @property
     def _source_subfolder(self):
         return os.path.join(self.source_folder, "source_subfolder")
 
     @property
     def _nodejs_arch(self):
-        if str(self.settings.os) == "Linux":
-            if str(self.settings.arch).startswith("armv7"):
+        if str(self.info.settings.os) == "Linux":
```

Comment:
`self.info` should only be used in `package_info` this looks very bizarre 🤔

Reply:
See the reason here: https://github.com/conan-io/conan/issues/13506 Basically it is a bug in conan itself ;)

Reply:
No no, it's usually a sign of doing something funky that probably has a different solution.

There are checks that need to happen in both generate and package_id